src/jsn.erl

```diff
@@ -656,6 +686,38 @@ from_proplist(X, _Options) -> X.
 %% internal functions
 %%==============================================================================
 
+-spec apply_condition(condition()|conditions(), json_term()) -> boolean().
+apply_condition({Path, ConditionFun}, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(get(Path, Element));
+apply_condition({Path, Value}, Element) ->
+    get(Path, Element) == Value;
+apply_condition(ConditionFun, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(Element);
+apply_condition([Condition|Rest], Element) ->
+    case apply_condition(Condition, Element) of
+        true ->
+            apply_condition(Rest, Element);
+        false ->
+            false
+    end;
+apply_condition([], _Element) ->
+    true;
+apply_condition(_, _) ->
+    erlang:error(invalid_condition).
+
+
+-spec apply_selection(selection()|selections(), json_term()) -> term().
+apply_selection(identity, Element) ->
+    Element;
+apply_selection({value, Path}, Element) ->
+    get(Path, Element);
+apply_selection({value, Path, Default}, Element) ->
+    get(Path, Element, Default);
+apply_selection(Selections, Element) when is_list(Selections) ->
+    [apply_selection(Selection, Element) || Selection <- Selections];
+apply_selection(_, _) ->
+    erlang:error(invalid_output_spec).
```

Comment:
Same issue here, more or less (badarg vs. custom error), but also, what is an `output_spec`? Shouldn't that say `selection` if we're using a custom error? I don't think `output_spec` term was used anywhere else in the code.

Reply:
Agree this error can be better stated as an `invalid_selection`